### PR TITLE
Install foundry on the container if required

### PR DESCRIPTION
**Pull request by elopez**
If the target is a directory and it contains a foundry.toml file,
install foundry using foundryup before invoking slither. Also run
foundry install to gather any required dependencies.

Fixes: #5

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -64,6 +64,25 @@ install_node()
     popd >/dev/null
 }
 
+install_foundry()
+{
+    if [[ -d "$TARGET" ]] && [[ -f "$TARGET/foundry.toml" ]]; then
+        echo "[-] Foundry target detected, installing foundry nightly"
+
+        wget -q -O foundryup https://raw.githubusercontent.com/foundry-rs/foundry/36a66c857ff148f6ed007cdcd3402077de3595cf/foundryup/foundryup
+        if [ ! "c0c6711dc39deae65bebcc0320fec1265930895a527e18daa643a708218c07ae  foundryup" = "$(sha256sum foundryup)" ]; then
```
Comment (gakonst) [inline entrypoint.sh:73]: Nice!

**Review by gakonst — APPROVED** [review]
LGTM - anything else needed to get this over the line?

**Comment by elopez** [comment]
@gakonst it's missing a way to pass a different foundry version -- I saw the foundry action offers an option to change that; how important do you reckon it is to have something like that in the Slither action? We can always add it later if it's non-urgent.

**Comment by gakonst** [comment]
I think it's not really urgent, we've had a culture of using nightlies for the most part for Foundry so I'd say let's go ahead, and iterate, given it'd unblock the action from being used in 3rd party CI

**Pull request merged**